backend/src/main/java/de/jankahle/capstone/service/RecipeTextFilterService.java

```diff
@@ -0,0 +1,97 @@
+package de.jankahle.capstone.service;
+
+import de.jankahle.capstone.model.Ingredient;
+import de.jankahle.capstone.model.Recipe;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.regex.*;
+
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class RecipeTextFilterService {
+
+    public Recipe parseStringToRecipe(String recipeString) {
+        return Recipe.builder().build();
+    }
+
+    Optional<Ingredient> parseIngredient(String recipeLine) {
+
+        ArrayList<String> matcherResults = filterLineForIngredientDetails(recipeLine);
+
+        if (matcherResults.isEmpty())
+            return Optional.empty();
+
+        String matcherResult = normalizeMatcherResults(matcherResults);
+
+        String[] ingredientData = matcherResult.split(" ");
```

Comment:
you may want to consider using a `List<String>` as it is more flexible down the road and could e.g. be used in streams, etc.

Reply:
I do not see this as necessary, ingredientData is only used in this context and not expanded later on. 
I am generally a fan of lists, but in this case this would be overkill as the array will always be downsized to 3